test/src/unit-cppapi-vfs.cc

```diff
@@ -85,15 +87,34 @@ TEST_CASE("C++ API: Test VFS ls", "[cppapi][cppapi-vfs][cppapi-vfs-ls]") {
   file2 = tiledb::sm::path_win::uri_from_path(file2);
   subdir = tiledb::sm::path_win::uri_from_path(subdir);
   subdir2 = tiledb::sm::path_win::uri_from_path(subdir2);
+  subdir_empty = tiledb::sm::path_win::uri_from_path(subdir_empty);
 #endif
 
   // Check results
   std::sort(children.begin(), children.end());
-  REQUIRE(children.size() == 4);
+  REQUIRE(children.size() == 5);
   CHECK(children[0] == file);
   CHECK(children[1] == file2);
   CHECK(children[2] == subdir);
   CHECK(children[3] == subdir2);
+  CHECK(children[4] == subdir_empty);
+
+#ifndef _WIN32
+  // List recursive (not supported for Windows yet)
+  // (this is tested more exhaustively in
+  // `tiledb/sm/filesystem/test/unit_ls_filtered.cc`)
+  std::vector<std::string> children_recursive = vfs.ls_recursive(dir);
```

Comment:
I reverted the external API changes since they were not done correctly (causing failures in some unit tests), and as we discussed this morning I think more substantive changes are needed than belong in this more-tightly-scoped PR.